src/api/router/Router.ts

```diff
@@ -0,0 +1,115 @@
+import { NavigationView } from "tabris";
+import { ListLike, Mutation } from "../List";
+import { RouterMatcher } from "./RouterMatcher";
+import { RouterHistory, HistoryItem } from "./RouterHistory";
+import { Route } from "./Route";
+import { Constructor } from "../../internals/utils";
+
+export type RouterConfig = {
+  name: string;
+  route: Constructor<Route>
+};
+
+export type RouterProperties = {
+  navigationView: NavigationView,
+  routers?: ListLike<RouterConfig>,
+  defaultRoute?: HistoryItem,
+  history?: ListLike<HistoryItem>
+};
+
+export class Router<ItemType extends  HistoryItem = HistoryItem> {
+
+  private _navigationView: NavigationView;
+  private _routes: ListLike<RouterConfig>;
+
+  private _routerHistoryObserver: RouterHistory;
+  private _routerMatcher: RouterMatcher;
+
+  constructor({navigationView, routers, defaultRoute, history} : RouterProperties) {
```

Comment:
hmmm.... do we need a default route? What's the use-case?

Reply:
I think it's useful to navigate to the initial route